patches/systemd/0002-Add-support-for-systemd-boot-for-Managarm.patch

```diff
@@ -0,0 +1,651 @@
+diff --git a/src/core/manager.c b/src/core/manager.c
+index 22b1822..3b74cb0 100644
+--- a/src/core/manager.c
++++ b/src/core/manager.c
+@@ -926,6 +926,7 @@ int manager_new(RuntimeScope runtime_scope, ManagerTestRunFlags test_run_flags,
+                 .executor_fd = -EBADF,
+         };
+ 
++
+         unit_defaults_init(&m->defaults, runtime_scope);
+ 
+ #if ENABLE_EFI
+diff --git a/src/core/mount.c b/src/core/mount.c
+index 66917df..82fa9dd 100644
+--- a/src/core/mount.c
++++ b/src/core/mount.c
+@@ -1553,6 +1553,11 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
+                         f == MOUNT_SUCCESS,
+                         code, status);
+ 
++#ifdef __managarm__
++        if(m->state == MOUNT_MOUNTING && f == MOUNT_SUCCESS)
++                m->state = MOUNT_MOUNTING_DONE;
++#endif
++
+         /* Note that due to the io event priority logic, we can be sure the new mountinfo is loaded
+          * before we process the SIGCHLD for the mount command. */
+ 
+@@ -2124,6 +2129,9 @@ static int drain_libmount(Manager *m) {
+ }
+ 
+ static int mount_process_proc_self_mountinfo(Manager *m) {
++#ifdef __managarm__
++        return 0;
++#else
+         _cleanup_set_free_ Set *around = NULL, *gone = NULL;
+         const char *what;
+         int r;
+@@ -2236,6 +2244,7 @@ static int mount_process_proc_self_mountinfo(Manager *m) {
+         }
+ 
+         return 0;
++#endif
+ }
+ 
+ static int mount_dispatch_io(sd_event_source *source, int fd, uint32_t revents, void *userdata) {
+diff --git a/src/journal/journald-native.c b/src/journal/journald-native.c
+index 0600102..e4d3be2 100644
+--- a/src/journal/journald-native.c
++++ b/src/journal/journald-native.c
+@@ -493,9 +493,11 @@ int server_open_native_socket(Server *s, const char *native_socket) {
+                         log_warning_errno(r, "SO_PASSSEC failed: %m");
+         }
+ 
++#ifndef __managarm__
+         r = setsockopt_int(s->native_fd, SOL_SOCKET, SO_TIMESTAMP, true);
+         if (r < 0)
+                 return log_error_errno(r, "SO_TIMESTAMP failed: %m");
++#endif
+ 
+         r = sd_event_add_io(s->event, &s->native_event_source, s->native_fd, EPOLLIN, server_process_datagram, s);
+         if (r < 0)
+diff --git a/src/journal/journald-server.c b/src/journal/journald-server.c
+index 07748f0..c0ccd79 100644
+--- a/src/journal/journald-server.c
++++ b/src/journal/journald-server.c
+@@ -162,9 +162,11 @@ static int cache_space_refresh(Server *s, JournalStorage *storage) {
+         if (space->timestamp != 0 && usec_add(space->timestamp, RECHECK_SPACE_USEC) > ts)
+                 return 0;
+ 
+-        r = server_determine_path_usage(s, storage->path, &vfs_used, &vfs_avail);
+-        if (r < 0)
+-                return r;
++        // r = server_determine_path_usage(s, storage->path, &vfs_used, &vfs_avail);
++        // if (r < 0)
++        //         return r;
++        vfs_used = 0;
++        vfs_avail = 16384;
```

Comment:
Why is this done? This should also get a comment.

Reply:
I've added a comment.